chain/chain/src/chain.rs

```diff
@@ -2370,6 +2370,30 @@ impl Chain {
     pub fn is_chunk_orphan(&self, hash: &CryptoHash) -> bool {
         self.blocks_with_missing_chunks.contains(hash)
     }
+
+    /// Check if can sync with sync_hash
+    pub fn sync_hash_validity(&mut self, sync_hash: &CryptoHash) -> Result<bool, Error> {
+        let head = self.head()?;
+        match self.get_block_header(sync_hash) {
+            Ok(sync_header) => {
+                let sync_header = sync_header.clone();
+                // The Epoch of sync_hash may be either the current one or the previous one
+                if head.epoch_id == *sync_header.epoch_id()
+                    || head.epoch_id == *sync_header.next_epoch_id()
+                {
+                    if self.runtime_adapter.is_next_block_epoch_start(sync_header.prev_hash())? {
+                        self.get_block(sync_hash)?;
+                        Ok(true)
+                    } else {
+                        Ok(false) // sync_hash is not on the Epoch boundary
+                    }
+                } else {
+                    Ok(false) // invalid Epoch of sync_hash, possible malicious behavior
+                }
+            }
+            Err(_) => Ok(false), // couldn't find header for sync_hash, possible malicious behavior
```

Comment:
Match against DBNotFound here, it's weird to not propagate other errors, if the function returns Result.

Reply:
++